### ISSUE TITLE
Non-animated gifs on Imgur fail with 'Sorry, video reproduction failed'.

### ISSUE THREAD

**Issue by SA-Flipperwaldt** [SpartanJ/ImgurViewer#70]
Example: [https://i.imgur.com/Pq1wyvp.gif](https://i.imgur.com/Pq1wyvp.gif)

The same file hosted elsewhere works fine. Simply changing the file extension in the Imgur link to png or jpg works, but I can't police what other people post.. Changing it to gifv doesn't, which might be what the bandwidth saving feature does?

I don't know that it is possible to tell animated and non-animated gifs apart ahead of time, so I don't know what a proper solution is. Optional retry with the link as given on error?

Anyway, thanks for your time so far.

**Comment by SpartanJ** [comment]
Yes, the issue is what you mention, I cannot tell ahead of time if the gif is a static image, so ImgurViewer assumes (incorrectly) that the image is animated (gifs are mostly used for animated images these days). I could probably fallback if the reproduction fails. I'm currently not working actively on the application because I'm focusing on other projects in my free time. So I'll try to fix all the most severe bugs first the next time I touch the project, but I won't probably add new features. Android keeps getting worst with each release and got tired of Android development (you'll see some bug reports with Android 13, that's what happens with each new version of Android, they literally break the applications and expect the devs to fix them).

**Comment by SA-Flipperwaldt** [comment]
No worries. Your time is your own. Thanks for updating me on what to realistcally expect.
